pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -127,7 +127,7 @@ public void runJob() {
       // Deregister the main thread and wait for all threads done
       phaser.awaitAdvance(phaser.arriveAndDeregister());
     }
-    CombineOperatorUtils.setExecutionStatistics(mergedBlock, _operators, totalWorkerThreadCpuTimeNs.get());
+    CombineOperatorUtils.setExecutionStatistics(mergedBlock, _operators, totalWorkerThreadCpuTimeNs.get(), _numThreads);
```

Comment:
Let's say there are 8 tasks and 4 threads in the executor service. At any point in time, the executor service will run 4 worker threads/tasks on CPU assuming numCores >= 4

So, let's say following happens
task 1 - thread 1 -> threadCpuTime - t1
task 2 - thread 2 -> threadCpuTime - t2
task 3 - thread 3 -> threadCpuTime - t3
task 4 - thread 4 -> threadCpuTime - t4

task 5 - thread 1 -> threadCpuTime - t5
task 6 - thread 2 -> threadCpuTime - t6
task 7 - thread 3 -> threadCpuTime - t7
task 8 - thread 4 -> threadCpuTime - t8

Today, we compute the total as sum (t1... t8) to get total threadCpuTime across all query threads.
So, in the adjustment equation when we do threadCpuTime / N , should N be used as 8 or 4

I think N should be 4 because number of tasks is dependent on the number of operators (segments)

GroupBy operator passes _numThreads as numOperators to BaseCombineOperator

Reply:
So, let's change _numThreads to numTasks or something more meaningful

We have 2 cases

- GroupBy passes _numTasks as numOperators
- Selection and everyone else passes _numTasks as a factor of number of cores see function  `public static int getNumThreadsForQuery(int numOperators)` in CombineOperatorUtils
- 
case 1 - query on 100 segments with 8 threads in worker executor service

- group by - _numTasks = 100, _numOperators = 100, worker threads = 8
- selection and everyone else - _numTasks = 8, _numOperators = 100, worker threads = 8

case 2 - query on 4 segments with 8 threads in worker executor service

- group by - _numTasks = 4, _numOperators = 4, worker threads = 8
- selection and everyone else - _numTasks = 4, _numOperators = 4, worker threads = 8

Let's have good tests for all these. You can also run OfflineClusterIntegrationTest in debug mode and see how _numOperators, _numTasks and numWorkerThreads look like